Samples/src/IntroToIQSharp/Notebook.ipynb

```diff
@@ -56,138 +90,425 @@
-    "\n",
-    "For example, you can create a new operation that calls the previously compiled `HelloQ`:\n"
+    "To verify this, you can use the `%simulate` magic to simulate the execution of the operation in a QPU:"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": 3,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "outputs": [
+    {
+     "data": {
+      "application/json": "0",
+      "text/plain": [
+       "Zero"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%simulate HelloQuantum"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "notes"
+    }
+   },
+   "source": [
+    "A more interesting example for a Q# operation is a quantum random number generator (QNRG). A QNRG returns a different value each time it is invoked:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "outputs": [
+    {
+     "data": {
+      "application/json": "[\"QNRG\"]",
+      "text/html": [
+       "<ul><li>QNRG</li></ul>"
+      ],
+      "text/plain": [
+       "QNRG"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "open Microsoft.Quantum.Diagnostics;\n",
+    "    \n",
+    "operation QNRG() : Result {\n",
+    "    using (q = Qubit()) {\n",
+    "        H(q);\n",
+    "        let r = M(q);\n",
+    "        \n",
+    "        Reset(q);\n",
+    "        return r;\n",
+    "    }\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
    "metadata": {},
    "outputs": [
     {
      "data": {
-      "application/json": "[\"HelloAndres\"]",
+      "application/json": "1",
+      "text/plain": [
+       "One"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%simulate QNRG"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "notes"
+    }
+   },
+   "source": [
+    "As mentioned, one of the key characteristics of a qubit is that during execution it's state represents the probabiliy of measuring $|0\\rangle$ or $|1\\rangle$. In particular, the [`H`](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.primitive.h) primitive operation modifies the state of a qubit from $|0\\rangle$ to a state in which there is 50% probability of measuring  $|0\\rangle$ and 50% probability of measuring  $|1\\rangle$.\n",
+    "\n",
+    "> When the state of a qubit is such that it is not a 100% $|0\\rangle$ or 100% $|1\\rangle$, \n",
+    "> we say the qubit is in **superposition**."
```

Comment:
Superposition is inherently a basis-dependent concept, for example |0⟩ is an equal superposition of |+⟩ and |–⟩. It may be helpful to clarify that here:
```suggestion
    "> we say the qubit is in **superposition** of $|0\\rangle$ and $|1\\rangle$."
```